### ISSUE TITLE
How to access to the cv_score of the fitted pipeline

### ISSUE THREAD

**Issue by ArioBattolla** [EpistasisLab/tpot#886]
I am doing regression for the Boston Housing Price dataset using TPOTRegressor.
Since I am comparing the score of TPOTRegressor with other models, defined by hand, I would like to access to the cv_score of the best model, the one saved as "fitted_pipeline_". How can I access it?
Because with TPOTRegressor function, I can define the cross-validation and I want to get the score of this cross-validation.
I saw that the attribute "evaluated_individuals_" allows to look at the "internal_cv_score"of all models, but I would like to know if there is an easier and faster way to get the best cv_score, something like: tpot.best_cv_score
How can I do that?
Thank you very much

**Comment by weixuanfu** [comment]
You could use the [`cross_val_score`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html) to reproduce the **average** cv score in TPOT. A demo below:

```python
tpot_obj = TPOTRegessror(cv=5, scoring='r2')
tpot_obj.fit(X, y)
sklearn_pipeline = tpot_obj.fitted_pipeline_
# reset random state of all operators in the pipeline 
tpot_obj._set_param_recursive(sklearn_pipeline.steps, 'random_state', 42)
cv_scores = cross_val_score(sklearn_pipeline, X, y, cv=5, scoring='r2', verbose=0)
mean_cv_scores = np.mean(cv_scores)
```

**Comment by ArioBattolla** [comment]
Yes ok, but this is inefficient, since you have already computed that value during tpot_obj.fit(X, y).
I wanted to avoid further computations and get it easily. But I do not know if this value is stored somewhere during the fitting. If it is not, I think your way is the only solution

**Comment by weixuanfu** [comment]
Oh, I just recall that `tpot_obj._optimized_pipeline_score` is the average cv score for `tpot_obj.fitted_pipeline_`.

**Comment by ArioBattolla** [comment]
Great!! This is what I was looking for.
Thank you very much :)

**Comment by ArioBattolla** [comment]
Is there a way also to have the standard deviation of the scores?
Because I guess that this value `tpot_obj._optimized_pipeline_score` is the mean value.
Thank you

**Comment by weixuanfu** [comment]
No, TPOT doesn't estimate this standard deviation.

**Comment by ArioBattolla** [comment]
Ah okok thank you!
So I guess I need to proceed in this way you suggested me few comments ago.
Can I ask you what is the purpose of `tpot_obj._set_param_recursive(sklearn_pipeline.steps, 'random_state', 42)`, why do we need to reset the random_state? 
Thank you very much

> You could use the [`cross_val_score`](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_score.html) to reproduce the **average** cv score in TPOT. A demo below:
> 
> ```python
> tpot_obj = TPOTRegessror(cv=5, scoring='r2')
> tpot_obj.fit(X, y)
> sklearn_pipeline = tpot_obj.fitted_pipeline_
> # reset random state of all operators in the pipeline 
> tpot_obj._set_param_recursive(sklearn_pipeline.steps, 'random_state', 42)
> cv_scores = cross_val_score(sklearn_pipeline, X, y, cv=5, scoring='r2', verbose=0)
> mean_cv_scores = np.mean(cv_scores)
> ```
